src/kademlia/dht_state.cpp

```diff
@@ -36,16 +36,44 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <libtorrent/socket_io.hpp>
 
 namespace libtorrent { namespace dht {
-namespace {
 
-	node_id extract_node_id(bdecode_node const& e, string_view key)
+	node_ids_t extract_node_ids(bdecode_node const& e, string_view key)
 	{
-		if (e.type() != bdecode_node::dict_t) return node_id();
-		auto nid = e.dict_find_string_value(key);
-		if (nid.size() != 20) return node_id();
-		return node_id(nid);
+		if (e.type() != bdecode_node::dict_t) return node_ids_t();
+		node_ids_t ret;
+		// first look for an old-style nid
```

Comment:
how is that possible that in this DHT's state, there is an old-style nid? shouldn't that be an error?